File: 9.palindrome-number.java
/*
 * @lc app=leetcode id=9 lang=java
 *
 * [9] Palindrome Number
 */

// @lc code=start
class Solution {
    public boolean isPalindrome(int x) {
        String s = Integer.toString(x);
        int l = s.length();
        for(int i=0; i<l; i++){
            if(s.charAt(i) != s.charAt(l-1-i)){
                return false;
            }
        }
        return true;
    }
}
// @lc code=end

File: Golang/9.palindrome-number.go
/*
 * @lc app=leetcode id=9 lang=golang
 *
 * [9] Palindrome Number
 */

// @lc code=start
func isPalindrome(x int) bool {
    var reversedNum int
	if x<0||(x%10==0&&x!=0){
		return false
	}
	for x>reversedNum{
		reversedNum = reversedNum*10+x%10
		x=x/10
	}
	return x==reversedNum||x==reversedNum/10
}
// @lc code=end

